cartographer/mapping_2d/sparse_pose_graph.cc

```diff
@@ -444,23 +438,25 @@ void SparsePoseGraph::RunOptimization() {
   const auto& node_data = optimization_problem_.node_data();
   for (int trajectory_id = 0;
        trajectory_id != static_cast<int>(node_data.size()); ++trajectory_id) {
-    int node_data_index = 0;
     const int num_nodes = trajectory_nodes_.num_indices(trajectory_id);
-    int node_index = optimization_problem_.num_trimmed_nodes(trajectory_id);
-    for (; node_data_index != static_cast<int>(node_data[trajectory_id].size());
-         ++node_data_index, ++node_index) {
-      const mapping::NodeId node_id{trajectory_id, node_index};
-      trajectory_nodes_.at(node_id).pose = transform::Embed3D(
-          node_data[trajectory_id][node_data_index].point_cloud_pose);
+    int last_node_data_index = 0;
```

Comment:
Move down where it is used. `last_optimized_node_index`?

Reply:
It is being assigned in right after the declaration.([here](https://github.com/magazino/cartographer/blob/use_map_trajectory_node/cartographer/mapping_2d/sparse_pose_graph.cc#L448)). Or you think it makes sense to assign it right before [this line](https://github.com/magazino/cartographer/blob/use_map_trajectory_node/cartographer/mapping_2d/sparse_pose_graph.cc#L458) with `node_data.at(trajectory_id).rbegin()->first`?

Reply:
nevermind. you have answered below.